### DISCUSSION TITLE
primevue/splitter: There is a huge bug or rather a very necessary requirement here.

### DISCUSSION THREAD

**Discussion by zhanghengxin** [primefaces/community#434] [PrimeVue] [answered]
# primevue/splitter: There is a huge bug or rather a very necessary requirement here.
@cagataycivici @onursenture @tandraschko @melloware 

- Firstly, I pursue a clean and tidy feeling, so my choice is to use Tailwind with PrimeVUE without styles.

- Then this is my dome repository: https://github.com/lit-oppai/raid-ts/commit/9b599c6249609e83c858fbacc6d25571241805d0

My goal is: 1. Always use Tailwind to write style parts. 2. Use splitter to split pages. 3. On large pages, use the splitter function normally and set the initial size to left 20%, right 80%, and it can be dragged. However, when on mobile, I want to lock the left width.

- Here is the problem, below the md breakpoint, I can lock the width by setting styles, but the unstyled Prime will use flex to leave 100% - 80% - left side value set as empty space.

![image](https://github.com/primefaces/community/assets/24197448/3dbcfcda-d4b5-4df4-9e7d-a91b6ffa0d60)

Now my hope is (any of):

- Can directly simply override through Tailwind while retaining the dragging functionality of the splitter component.

- (Not a very preferred solution) Let me pass values in JS to tell the component when the dragging functionality becomes invalid.

- Inform the component of breakpoint functionality, i.e.: at what breakpoint to set what width.

**Comment by zhanghengxin** [comment]
The current way i am handling it is to add a flex-grow:1 key-value to right part.

**Reply by zhanghengxin** [reply]
This way of solving it uses overriding styles, but the component's behavior should not be mutually influenced by CSS. Currently I do not have a very good recommendation.

> Rerhaps it is to change the style to style="flex-grow: 1;" in the unstyled mode.

**Reply by zhanghengxin** [reply]
i have thought about it, and gotten rid of the best answer. this kind of fix does not satisfy the three settings i mentioned above. above all，it still has not been resolved internally,but by overriding.

**Comment by tandraschko** [comment]
Could you use the PrimeVue channel, not PrimeFaces?

**Reply by zhanghengxin** [reply]
It has been changed, but I think this is a design problem because its dragging directly modifies the style, and the scenarios and processes of this part should be supplemented.

so, it should belong to primeFaces.

**Reply by melloware** [reply]
@zhanghengxin if you know a fix then please open a GitHub issue and submit a PR to PrimeVue.

**Reply by zhanghengxin** [reply]
As you wish.
https://github.com/primefaces/primevue/issues/4794
https://github.com/primefaces/primevue/pull/4801

I chose a very crude way, which may need to be improved a few times under your guidance as required.
What's happened recently? I checked the commit history and found there were lots of commits made in a short period of time.

**Comment by zhanghengxin** [comment, marked as answer]
https://github.com/primefaces/primevue/pull/4801
